### DISCUSSION TITLE
Alternative way to access @this in JS

### DISCUSSION THREAD

**Discussion by ph55** [livewire/livewire#2028] [1. Help] [answered]
Is there any other way to access livewire's `@this` object in JS ?
I'm using IntelliJ PHPStorm and it highlight this as error:

![image](https://user-images.githubusercontent.com/1386622/99149726-8731bb80-2698-11eb-98f1-cef2a0d19fba.png)

**Comment by ph55** [comment, marked as answer]
Ok.
That was quick lol.

Using `window.@this` solves the issue:
![image](https://user-images.githubusercontent.com/1386622/99149912-a41abe80-2699-11eb-863f-51fadb17e0cb.png)


Because livewire resolves it to following:
![image](https://user-images.githubusercontent.com/1386622/99149920-b694f800-2699-11eb-8b83-95687228391e.png)

And the window property of a Window object points to the window object itself. Thus, the following expressions all return the same window object:

```
window.window
window.window.window
window.window.window.window
// ...
```

[Source](https://developer.mozilla.org/en-US/docs/Web/API/Window/window
)

**Comment by psos** [comment]
@this is bad idea, becase it it's blade macro for laravel compontent class echoing javascript code:
window.Livewire.find('idOfCurrentLivewireCompontent')

I would prefer use global hooks:
https://laravel-livewire.com/docs/2.x/lifecycle-hooks

```js
<script>
    document.addEventListener("DOMContentLoaded", () => {
        Livewire.hook('component.initialized', (component) => {
                   ....
        })
        Livewire.hook('element.initialized', (el, component) => {})
        Livewire.hook('element.updating', (fromEl, toEl, component) => {})
        Livewire.hook('element.updated', (el, component) => {})
        Livewire.hook('element.removed', (el, component) => {})
        Livewire.hook('message.sent', (message, component) => {})
        Livewire.hook('message.failed', (message, component) => {})
        Livewire.hook('message.received', (message, component) => {})
        Livewire.hook('message.processed', (message, component) => {})
    });
</script>
```

But therefore i think better is to use compontent 
```html
<div wire:init="reorder">
// livewirecomponent.blade.php ...
</div>
```
